### PR TITLE
added ansible module for pool and hpe nimble info. also fixed few bug…

### PR DESCRIPTION
**Pull request by ar-india**
changes:
- added support for hpe_nimble_info and hpe_nimble_pool
- fixed acr bug where in we wer not able to create acr if the apply_to was set to snapshot. 
- removed unused code from IG
- refactored snapshot delete logic

Signed-off-by: ar-india <alokranjan2005@gmail.com>

### CODE REVIEW

**Comment by ar-india** [comment]
Please review

File: ansible_collection/hpe/nimble/plugins/module_utils/hpe_nimble.py [hunk 1]
```diff
@@ -98,6 +98,9 @@ def remove_unchanged_or_null_args(server_resp, **kwargs):
             found_changed_list = False
             if len(value) == 0:
                 continue
+            if len(value) != len(server_value):
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/module_utils/hpe_nimble.py:101]: if no of items in list differs.. then list is anyway not same. hence no need to check further

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_access_control_record.py [hunk 2]
```diff
@@ -32,6 +32,7 @@
     required: False
     choices:
     - volume
+    - snapshot
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_access_control_record.py:35]: acr's 'apply_to' should support "vol" "snapshot" and "both".

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_access_control_record.py [hunk 3]
```diff
@@ -159,9 +162,10 @@ def delete_acr(
     params = utils.remove_null_args(**kwargs)
 
     try:
-        acr_resp = client_obj.access_control_records.get(vol_name=volume, initiator_group_name=initiator_group, **params)
-        if acr_resp is not None:
-            client_obj.access_control_records.delete(acr_resp.attrs.get("id"))
+        acr_list_resp = client_obj.access_control_records.list(vol_name=volume, initiator_group_name=initiator_group, **params)
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_access_control_record.py:165]: using list , hence renamed variable

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_initiator_group.py [hunk 4]
```diff
@@ -135,30 +135,6 @@
     client = None
 import ansible_collections.hpe.nimble.plugins.module_utils.hpe_nimble as utils
 
-def is_initiator_present(
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_initiator_group.py:138]: removed unused func

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot.py [hunk 5]
```diff
@@ -208,11 +208,8 @@ def delete_snapshot(
         if utils.is_null_or_empty(snap_resp):
             return (False, False, f"Snapshot '{snapshot_name}' cannot be deleted as it is not present in given volume '{vol_name}'.", {})
         else:
-            snap_resp = client_obj.snapshots.delete(id=snap_resp.attrs.get("id"))
-            # confirm
-            snap_resp = client_obj.snapshots.get(id=None, vol_name=vol_name, name=snapshot_name)
-            if utils.is_null_or_empty(snap_resp):
-                return (True, True, f"Deleted snapshot '{snapshot_name}' successfully.", {})
+            client_obj.snapshots.delete(id=snap_resp.attrs.get("id"))
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot.py:211]: refactored. no need to check the resp of delete. as if it fails exception will be thrown

**Review by ar-india — COMMENTED** [review]
changes
- removed info state
- converted subset o/p's for config,minimum and all subset to be displayed a s a list rather than dictionary

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+author:
+  - Alok Ranjan (@ar-india)
+description: Manage storage pools on HPE Nimble Storage group.
+module: hpe_nimble_pool
+options:
+  array_list:
+    required: False
+    type: list
+    description:
+    - List of arrays in the pool with detailed information. When create/update array list, only arrays' ID is required.
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py]: nit: List of arrays in the pool with detailed information. To create or update array list, only array IDs are required.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+    description:
+    - Indicates if dedupe is enabled by default for new volumes on this pool.
+  force:
+    required: False
+    type: bool
+    default: False
+    description:
+    - Forcibly delete the specified pool even if it contains deleted volumes whose space is being reclaimed.
+      Forcibly remove an array from array_list via an update operation even if the array is not reachable.
+      There should no volumes currently in the pool for the forced update operation to succeed.
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py]: nit: There should be no volumes in the pool for the force update operation to succeed.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+        return (False, False, "Create pool failed as pool name is not present.", {})
+
+    try:
+        pool_resp = client_obj.pools.get(id=None, name=pool_name)
+        if utils.is_null_or_empty(pool_resp):
+            params = utils.remove_null_args(**kwargs)
+            pool_resp = client_obj.pools.create(name=pool_name,
+                                                **params)
+            if pool_resp is not None:
+                return (True, True, "Created pool '%s' successfully." % pool_name, {})
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py]: nit: Please use 'F-strings' format syntax while constructing messages/strings.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+        else:
+            return (False, False, "Pool '%s' cannot be created as it is already present." % pool_name, {})
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py]: nit: Please use 'F-strings' format syntax while constructing messages/strings.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py [hunk 10]
```diff
@@ -0,0 +1,351 @@
...
+
+    if utils.is_null_or_empty(pool_resp):
+        return (False, False, "Update pool failed as pool name is not present.", {})
+
+    try:
+        pool_name = pool_resp.attrs.get("name")
+        changed_attrs_dict, params = utils.remove_unchanged_or_null_args(pool_resp, **kwargs)
+        if changed_attrs_dict.__len__() > 0:
+            pool_resp = client_obj.pools.update(id=pool_resp.attrs.get("id"), **params)
+            return (True, True, "Pool '%s' already present. Modified the following fields:" % pool_name, changed_attrs_dict)
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_pool.py]: nit: Please use 'F-strings' format syntax while constructing messages/strings. 

PLEASE UPDATE/FIX THIS IN ALL OTHER PLACES OF THE CODE.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py [hunk 11]
```diff
@@ -0,0 +1,1024 @@
...
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+author:
+  - Alok Ranjan (@ar-india)
+description:
+  - Collect information from a HPE Nimble Storage array running the NimOs operating system. By default, the module will collect basic
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py]: Nimble Operating System or NimOS are preferred. However you can just say "Collect information from a HPE Nimble Storage array" as NimOS is implicit.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py [hunk 12]
```diff
@@ -0,0 +1,1024 @@
...
+    information including array, groups config, protection template, protection schedules, snapshots, snapshot collection, volume collection
+    and volume counts. Additional information can be collected based on the configured set of arguements.
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py]: nit: set of arguments (typo)

**Review by suneeth51 — APPROVED** [review]
lgtm.. Please fix those minor comments.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py (hunk 11 above)
Comment (ar-india) [reply ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py]: ok.

**Pull request merged**